sklearn/calibration.py

```diff
@@ -311,145 +320,153 @@ def _more_tags(self):
         }
 
 
-class _CalibratedClassifier:
-    """Probability calibration with isotonic regression or sigmoid.
+def _get_predictions(clf_fitted, X, label_encoder_):
+    """Returns predictions for `X` and index of classes present in `X`.
 
-    It assumes that base_estimator has already been fit, and trains the
-    calibration on the input set of the fit function. Note that this class
-    should not be used as an estimator directly. Use CalibratedClassifierCV
-    with cv="prefit" instead.
+    For predicitons, `decision_function` method of the `clf_fitted` is used.
+    If this does not exist, `predict_proba` method used.
 
     Parameters
     ----------
-    base_estimator : instance BaseEstimator
-        The classifier whose output decision function needs to be calibrated
-        to offer more accurate predict_proba outputs. No default value since
-        it has to be an already fitted estimator.
+    clf_fitted : Estimator instance
+        Fitted classifier instance.
 
-    method : 'sigmoid' | 'isotonic'
-        The method to use for calibration. Can be 'sigmoid' which
-        corresponds to Platt's method or 'isotonic' which is a
-        non-parametric approach based on isotonic regression.
+    X : array-like
+        Sample data used for the predictions.
 
-    classes : array-like, shape (n_classes,), optional
-            Contains unique classes used to fit the base estimator.
-            if None, then classes is extracted from the given target values
-            in fit().
+    label_encoder_ : LabelEncoder instance
+        LabelEncoder instance fitted on all the targets.
 
-    See also
-    --------
-    CalibratedClassifierCV
+    Returns
+    -------
+    df : array-like, shape (X.shape[0], len(clf_fitted.classes_))
+        The predictions. Note array is of shape (X.shape[0], 1) when there are
+        2 classes.
 
-    References
-    ----------
-    .. [1] Obtaining calibrated probability estimates from decision trees
-           and naive Bayesian classifiers, B. Zadrozny & C. Elkan, ICML 2001
+    idx_pos_class : array-like, shape (n_classes,)
```

Comment:
```suggestion
    pos_class_indices : array-like, shape (n_classes,)
```

otherwise the name `idx` suggests it's a single value

Though it seems that in the binary case this isn't the `pos_class` index?

Reply:
Yes, in the binary case we only output the probability of `clf_fitted.classes[1]`:
```
    elif hasattr(clf_fitted, "predict_proba"):
        df = clf_fitted.predict_proba(X)
        if len(label_encoder_.classes_) == 2:
            df = df[:, 1:]
```
though this can be considered the  'positive class'?

`idx_pos_class` indicates which classes are 'present' in a particular train/test subset